src/parser.rs

```diff
@@ -166,51 +155,244 @@ impl<'a> DocParser<'a> {
-  ) -> Result<Vec<DocNode>, DocError> {
-    let doc_nodes =
-      self.parse_with_reexports_inner(specifier, HashSet::new())?;
-    self.collect_diagnostics_for_nodes(&doc_nodes);
-    Ok(doc_nodes)
+    reference_def: &ReferenceDef,
+    mut name_path: Vec<String>,
+    star: bool,
+  ) -> Result<Option<Vec<DocNode>>, DocError> {
+    let module = resolve_deno_graph_module(self.graph, specifier)?;
+
+    match module {
+      Module::Js(_) | Module::Json(_) | Module::Wasm(_) => {
+        let module_info = self.get_module_info(module.specifier())?;
+        let exports = module_info.exports(&self.root_symbol);
+        let root_name = name_path.remove(0);
+        let (mut export_name, export) = exports
+          .resolved
+          .iter()
+          .find(|(name, _)| &&root_name == name)
+          .unwrap();
+
+        let export = export.as_resolved_export();
+        let mut export_symbol = export.module.symbol(export.symbol_id).unwrap();
+
+        let mut definitions = self
+          .root_symbol
+          .go_to_definitions(export.module, export_symbol)
+          .collect::<Vec<_>>();
+
+        if let Some(first_def) = definitions.first() {
+          use deno_graph::symbols::DefinitionKind;
+          match first_def.kind {
+            DefinitionKind::ExportStar(file_dep) => {
+              debug_assert_eq!(definitions.len(), 1);
+              let specifier = self.resolve_dependency(
+                &file_dep.specifier,
+                first_def.module.specifier(),
+              )?;
+              return self.resolve_dangling_reference(
+                specifier,
+                reference_def,
+                name_path,
+                true,
+              );
+            }
+            DefinitionKind::Definition => {
+              for (i, name_path_item) in name_path.iter().enumerate() {
+                let Some((name, id)) = export_symbol
+                  .exports()
+                  .iter()
+                  .find(|(name, _)| &name_path_item == name)
+                else {
+                  let definitions = self
+                    .root_symbol
+                    .go_to_definitions(export.module, export_symbol)
+                    .collect::<Vec<_>>();
+
+                  return self.resolve_dangling_reference(
+                    definitions.first().unwrap().module.specifier(),
+                    reference_def,
+                    // -1 to include the root
+                    name_path[i - 1..].to_vec(),
+                    false,
+                  );
+                };
+
+                export_name = name;
+                export_symbol = export.module.symbol(*id).unwrap();
+              }
+
+              definitions = self
+                .root_symbol
+                .go_to_definitions(export.module, export_symbol)
+                .collect::<Vec<_>>();
+            }
+          }
+        }
+
+        let original_range = &export_symbol.decls().first().unwrap().range;
```

Comment:
What happens if it refers to multiple declarations? What if those declarations are found in different files? Do we have a test for that?

Reply:
that should be handled by tests/specs/namespace_reexport_member.txt